apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php

```diff
@@ -17,6 +17,13 @@ function should_show_coming_soon_page() {
 		return false;
 	}
 
+	$share_code = get_share_code();
+	if ( is_accessed_by_valid_share_link( $share_code ) ) {
+		track_preview_link_event();
+		setcookie( 'wp_share_code', $share_code, time() + 3600, '/', false, is_ssl() );
+		return false;
```

Comment:
What do you think about redirecting after the cookie is set, so we discard the query param from the URL?

Reply:
What would be the benefit of discarding the parameter?

Reply:
> What would be the benefit of discarding the parameter?

A cleaner URL. No functional benefit.